core/app/[locale]/(default)/(auth)/login/_components/login-form.tsx

```diff
@@ -1,132 +0,0 @@
-'use client';
-
-import { useTranslations } from 'next-intl';
-import { ChangeEvent, useActionState, useState } from 'react';
-import { useFormStatus } from 'react-dom';
-
-import { Link } from '~/components/link';
-import { Button } from '~/components/ui/button';
-import {
-  Field,
-  FieldControl,
-  FieldLabel,
-  FieldMessage,
-  Form,
-  FormSubmit,
-  Input,
-} from '~/components/ui/form';
-import { Message } from '~/components/ui/message';
-
-import { useAccountStatusContext } from '../../../account/(tabs)/_components/account-status-provider';
-import { login } from '../_actions/login';
-
-const SubmitButton = () => {
-  const { pending } = useFormStatus();
-  const t = useTranslations('Login');
-
-  return (
-    <Button
-      className="md:w-auto"
-      loading={pending}
-      loadingText={t('Form.submitting')}
-      variant="primary"
-    >
-      {t('Form.logIn')}
-    </Button>
-  );
-};
-
-export const LoginForm = () => {
-  const t = useTranslations('Login');
-
-  const [isEmailValid, setIsEmailValid] = useState(true);
-  const [isPasswordValid, setIsPasswordValid] = useState(true);
-  const [state, formAction] = useActionState(login, { status: 'idle' });
-  const { accountState } = useAccountStatusContext();
-
-  const isFormInvalid = state?.status === 'error';
-
-  const handleInputValidation = (e: ChangeEvent<HTMLInputElement>) => {
-    const validationStatus = e.target.validity.valueMissing;
-
-    switch (e.target.name) {
-      case 'email': {
-        setIsEmailValid(!validationStatus);
-
-        return;
-      }
-
-      case 'password': {
-        setIsPasswordValid(!validationStatus);
-      }
-    }
-  };
-
-  return (
-    <>
-      {accountState.status === 'success' && (
-        <Message className="col-span-full mb-8 w-full text-gray-500" variant={accountState.status}>
-          <p>{accountState.message}</p>
```

Comment:
Do we need to keep this behavior around in our new form? Should we send the account state message from the action `submission.reply` and show that on successful log-in? I'm guessing this is used to give the user some visual feedback if the login is successful while the action finishes redirecting?

Reply:
I feel like this is not necessary, if login is successful it should just redirect to the account page (or previous visited page before login, but that is another issue).

Reply:
Agreed 👍🏻